rct229/rules/section6/section6rule1.py

```diff
@@ -14,92 +19,135 @@ class Section6Rule1(RuleDefinitionListIndexedBase):
+                            allowable_LPD_space * space_p["floor_area"]
+                        )
 
             return {
-                "building_allowable_lighting_power": CalcQ(
-                    "electric_power", building_allowable_lighting_power
+                "allowable_LPD_BAM": CalcQ("power_density", allowable_LPD_BAM),
+                "building_segment_design_lighting_wattage": CalcQ(
+                    "electric_power", building_segment_design_lighting_wattage
                 ),
-                "building_design_lighting_power": CalcQ(
-                    "electric_power", building_design_lighting_power
+                "check_BAM_flag": check_BAM_flag,
+                "total_building_segment_area_p": CalcQ(
+                    "area", total_building_segment_area_p
+                ),
+                "allowable_lighting_wattage_SBS": CalcQ(
+                    "electric_power", allowable_lighting_wattage_SBS
                 ),
             }
 
         def rule_check(self, context, calc_vals=None, data=None):
-            return (
-                calc_vals["building_design_lighting_power"]
-                <= calc_vals["building_allowable_lighting_power"]
-            )
+            allowable_LPD_BAM = calc_vals["allowable_LPD_BAM"]
+            check_BAM_flag = calc_vals["check_BAM_flag"]
+            building_segment_design_lighting_wattage = calc_vals[
+                "building_segment_design_lighting_wattage"
+            ]
+            total_building_segment_area_p = calc_vals["total_building_segment_area_p"]
+            allowable_lighting_wattage_SBS = calc_vals["allowable_lighting_wattage_SBS"]
+
+            if allowable_LPD_BAM and not check_BAM_flag:
+                if building_segment_design_lighting_wattage <= max(
+                    allowable_LPD_BAM * total_building_segment_area_p,
+                    allowable_lighting_wattage_SBS,
+                ):
+                    return True
+                else:
+                    return False
+            elif not (allowable_LPD_BAM and check_BAM_flag):
+                if (
+                    building_segment_design_lighting_wattage
+                    <= allowable_lighting_wattage_SBS
+                ):
+                    return True
+                else:
+                    return False
+            elif allowable_LPD_BAM and check_BAM_flag:
+                if (
+                    building_segment_design_lighting_wattage
+                    <= allowable_LPD_BAM * total_building_segment_area_p
+                ):
+                    return True
+                else:
+                    return False
+            else:
+                return False
+
+        def get_pass_msg(self, context, calc_vals=None, data=None):
+            allowable_LPD_BAM = calc_vals["allowable_LPD_BAM"]
+            check_BAM_flag = calc_vals["check_BAM_flag"]
+
+            if allowable_LPD_BAM and not check_BAM_flag:
```

Comment:
Not sure if this is needed - The default message seems to be an empty string

Reply:
Fixed.